internal/cli/pipeline_inspect.go

```diff
@@ -111,10 +141,104 @@ func (c *PipelineInspectCommand) Run(args []string) int {
 		{
 			Name: "Root Step Name", Value: resp.RootStep,
 		},
-	}, terminal.WithInfoStyle())
+		{
+			Name: "Total Steps", Value: len(resp.Pipeline.Steps),
+		},
+	}
 
-	// TODO(briancain): Use graphviz to build a pipeline graph and display in the terminal?
+	s := uint64(c.flagRunSequence)
+	if s > 0 {
+		run, err := c.project.Client().GetPipelineRun(c.Ctx, &pb.GetPipelineRunRequest{
+			Pipeline: &pb.Ref_Pipeline{
+				Ref: &pb.Ref_Pipeline_Id{
+					Id: &pb.Ref_PipelineId{
+						Id: resp.Pipeline.Id,
+					},
+				},
+			},
+			Sequence: s,
+		})
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+		startJob, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
+			JobId: run.PipelineRun.Jobs[0].Id,
+		})
+		endJob, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
+			JobId: run.PipelineRun.Jobs[len(run.PipelineRun.Jobs)-1].Id,
+		})
+		var sha string
+		var msg string
+		if startJob.DataSourceRef != nil {
+			sha = startJob.DataSourceRef.Ref.(*pb.Job_DataSource_Ref_Git).Git.Commit
+			msg = startJob.DataSourceRef.Ref.(*pb.Job_DataSource_Ref_Git).Git.CommitMessage
+		}
+		output = append(output, []terminal.NamedValue{
+			{
+				Name: "Run Sequence", Value: run.PipelineRun.Sequence,
+			},
+			{
+				Name: "Jobs Queued", Value: run.PipelineRun.Jobs,
+			},
+			{
+				Name: "Run Started", Value: humanize.Time(startJob.QueueTime.AsTime()),
+			},
+			{
+				Name: "Run Completed", Value: humanize.Time(endJob.CompleteTime.AsTime()),
+			},
+			{
+				Name: "State", Value: run.PipelineRun.State,
+			},
+			{
+				Name: "Git Commit SHA", Value: sha,
+			},
+			{
+				Name: "Git Commit Message", Value: msg,
+			},
+		}...)
+	} else {
+		lastRun := runs.PipelineRuns[len(runs.PipelineRuns)-1]
+		startJob, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
+			JobId: lastRun.Jobs[0].Id,
+		})
+		endJob, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
```

Comment:
We should check this `err` and the `err` above it, otherwise use `_`?

Reply:
Ugh Github isn't picking up my changes, but this is addressed in https://github.com/hashicorp/waypoint/pull/3769/commits/4a8979b35e823775dbab26cc168859f884d2c7d2.